infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/configuration/impl/InfinispanRemoteConfiguration.java

```diff
@@ -87,4 +167,87 @@ public void initConfiguration(Map<?, ?> configurationMap, ServiceRegistryImpleme
 
 		log.tracef( "Initializing Infinispan Hot Rod client from configuration file at '%1$s'", configurationResource );
 	}
+
+	/**
+	 * Extract from the configuration the Hot Rod client properties: the one prefixed with
+	 * {@link InfinispanRemoteProperties#HOT_ROD_CLIENT_PREFIX} or defined in the resource file via
+	 * {@link InfinispanRemoteProperties#CONFIGURATION_RESOURCE_NAME}.
+	 * <p>
+	 * Note that the properties with the prefix will override the same properties in the resource file.
+	 */
+	private Properties getHotRodConfiguration(Map<?, ?> configurationMap, ConfigurationPropertyReader propertyReader, URL configurationResourceUrl) {
+		Properties hotRodConfiguration = new Properties();
+		loadResourceFile( configurationResourceUrl, hotRodConfiguration );
+		setAdditionalProperties( configurationMap, propertyReader, hotRodConfiguration );
+		setExpectedPropertiesIfNull( hotRodConfiguration );
+		validate( hotRodConfiguration );
+		return hotRodConfiguration;
+	}
+
+	/**
+	 * Load the properties from the resource file if one is specified
+	 */
+	private void loadResourceFile(URL configurationResourceUrl, Properties hotRodConfiguration) {
+		if ( configurationResourceUrl != null ) {
+			try ( InputStream openStream = configurationResourceUrl.openStream() ) {
+				hotRodConfiguration.load( openStream );
+			}
+			catch (IOException e) {
+				throw log.failedLoadingHotRodConfigurationProperties( e );
+			}
+		}
+	}
+
+	/**
+	 * Set the properties defined using the prefix {@link InfinispanRemoteProperties#HOT_ROD_CLIENT_PREFIX}
+	 *
+	 * @param configurationMap contains all the properties defined for OGM
+	 * @param propertyReader read the value of a property
+	 * @param hotRodConfiguration the Hot Rod configuration to update
+	 */
+	private void setAdditionalProperties(Map<?, ?> configurationMap, ConfigurationPropertyReader propertyReader, Properties hotRodConfiguration) {
+		// Programmatic properties override the resource file
+		for ( Entry<?, ?> property : configurationMap.entrySet() ) {
+			String key = (String) property.getKey();
+			if ( key.startsWith( HOT_ROD_CLIENT_PREFIX ) ) {
+				String hotRodProperty = key.substring( HOT_ROD_CLIENT_PREFIX.length() );
+				String value = propertyReader.property( key, String.class ).getValue();
+				if ( !ArrayHelper.contains( noPrefixProperties, hotRodProperty ) ) {
```

Comment:
I'd have used a hashmap here: much better "contains" efficiency.
Not important to change though, as it's a very small array (and only happens at boot time).